packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -83,7 +83,7 @@ class Observatory {
     if (isolates.isNotEmpty)
       return isolates.first;
 
-    _waitFirstIsolateCompleter = new Completer<IsolateRef>();
+    _waitFirstIsolateCompleter ??= new Completer<IsolateRef>();
```

Comment:
What would happen if `_waitFirstIsolateCompleter` was instantiated as a `final` field and never nulled out? Is this lazy instantiation an optimization?